src/Microsoft.DotNet.Build.Tasks/PackageFiles/init-tools.cmd

```diff
@@ -0,0 +1,34 @@
+@echo off
+setlocal
+
+set PROJECT_DIR=%~1
+set DOTNET_CMD=%~2
+set TOOLRUNTIME_DIR=%~3
+set TOOLS_DIR=%~dp0
+
+if not exist "%PROJECT_DIR%" (
+  echo ERROR: Cannot find project root path at [%PROJECT_DIR%]. Please pass in the source directory as the 1st parameter.
+  exit /b 1
+)
+
+if not exist "%TOOLS_DIR%" (
+  echo ERROR: Cannot find tools path at [%TOOLS_DIR%]. Please pass in the tools directory as the 2nd parameter.
+  exit /b 1
+)
+
+if not exist "%DOTNET_CMD%" (
+  echo ERROR: Cannot find dotnet command at [%DOTNET_CMD%]. Please pass in the dotnet command as the 3rd parameter.
+  exit /b 1
+)
+
+set tempDirectory=%cd%
```

Comment:
I'm find changing directories but I'm not sure why you need to do it this way. Why not just change and say there why do you need to go back? At any rate if you do need to do this you should use pushd and popd instead of trying to capture it this way.